lib/class-wp-rest-themes-controller.php

```diff
@@ -0,0 +1,233 @@
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+
+		$data = $this->add_additional_fields_to_object( $data, $request );
+		$data = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $theme ) );
+
+		/**
+		 * Filters theme data returned from the REST API.
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param WP_Theme         $theme    Theme object used to create response.
+		 * @param WP_REST_Request  $request  Request object.
+		 */
+		return apply_filters( 'rest_prepare_theme', $response, $theme, $request );
+	}
+
+	/**
+	 * Prepares links for the theme request.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_Theme $theme Theme object.
+	 * @return array Links for the given theme.
+	 */
+	protected function prepare_links( $theme ) {
+		$links = array(
+			'active'       => array(
+				'href' => rest_url( sprintf( '%s/%s/active', $this->namespace, $this->rest_base ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Retrieves the theme's schema, conforming to JSON Schema.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'theme',
+			'type'       => 'object',
+			'properties' => array(
+				'name'                 => array(
+					'description' => __( 'Theme name' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'stylesheet'                 => array(
+					'description' => __( 'The directory name of the theme.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'version'           => array(
+					'description' => __( 'Theme version' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'template'           => array(
+					'description' => __( 'The theme template' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'theme_supports' => array(
+					'description' => __( 'A list of features this theme supports.' ),
+					'type'        => 'array',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+					'properties'  => array(
+						'formats'      => array(
+							'description' => __( 'Post formats supported.' ),
+							'type'        => 'array',
+							'context'     => array( 'view' ),
+							'readonly'    => true,
+						),
+						'post-thumbnails' => array(
+							'description' => __( '' ),
+							'type'        => array( 'array', 'bool' ),
+							'context'     => array( 'view' ),
+							'readonly'    => true,
+						),
+					),
+				),
+			),
+		);
```

Comment:
Each of these schema attributes should be `context=edit`, as Theme objects are only accessed in the Edit context.